mlflow/models/signature.py

```diff
@@ -306,14 +306,24 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     if hints.input is None:
         return None
 
+    params = None
+    if isinstance(input_example, dict) and "params" in input_example:
+        params = input_example.pop("params")
+        if len(input_example) == 1:
+            input_example = list(input_example.values())[0]
```

Comment:
```suggestion
            input_example = next(iter(input_example.values()))
```

to skip constructing a list.

Reply:
The original code is also fine because `input_example` only has a small number of items.